editor/src/editor/nodes/camera.ts

```diff
@@ -72,19 +89,35 @@ export class EditorCamera extends FreeCamera {
 	public configureFromPreferences(): void {
 		try {
 			const keys = JSON.parse(localStorage.getItem("editor-camera-controls") as string);
-			this.keysUp = keys.keysUp;
-			this.keysDown = keys.keysDown;
-			this.keysLeft = keys.keysLeft;
-			this.keysRight = keys.keysRight;
-			this.keysUpward = keys.keysUpward;
-			this.keysDownward = keys.keysDownward;
+			if (keys && keys.keysUp && keys.keysDown && keys.keysLeft && keys.keysRight && keys.keysUpward && keys.keysDownward) {
+				this.keysUp = keys.keysUp;
+				this.keysDown = keys.keysDown;
+				this.keysLeft = keys.keysLeft;
+				this.keysRight = keys.keysRight;
+				this.keysUpward = keys.keysUpward;
+				this.keysDownward = keys.keysDownward;
+			} else {
+				// If preferences are incomplete, fall back to defaults
+				this.keysUp = EditorCamera.DEFAULT_KEYS.keysUp;
```

Comment:
This can be located in a method in the class like `private _setupDefaultKeys(): void { ... }`